indexing-service/src/main/java/org/apache/druid/indexing/common/task/batch/parallel/ParallelIndexSupervisorTask.java

```diff
@@ -491,6 +497,31 @@ private boolean isParallelMode()
     return isParallelMode(baseInputSource, ingestionSchema.getTuningConfig());
   }
 
+  /**
+   * Attempt to wait for indexed segments to become available on the cluster.
+   * @param reportsMap Map containing information with published segments that we are going to wait for.
+   */
+  private void waitForSegmentAvailability(Map<String, PushedSegmentsReport> reportsMap)
+  {
+    ArrayList<DataSegment> segmentsToWaitFor = new ArrayList<>();
+    reportsMap.values()
+              .forEach(report -> {
+                segmentsToWaitFor.addAll(report.getNewSegments());
+              });
+    ExecutorService availabilityExec = Execs.singleThreaded("ParallelTaskAvailabilityWaitExec");
+    try {
+      segmentAvailabilityConfirmationCompleted = waitForSegmentAvailability(
+          toolbox,
+          availabilityExec,
+          segmentsToWaitFor,
+          ingestionSchema.getTuningConfig().getAwaitSegmentAvailabilityTimeoutMillis()
```

Comment:
nit: Could we extract `ingestionSchema.getTuningConfig().getAwaitSegmentAvailabilityTimeoutMillis()` at class level if possible?

Reply:
I'm fine with that. addressed in latest commits